src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,419 @@
+                 L_rk_tol=1e-8):
+        if isinstance(samples_or_fom, TransferFunction) or hasattr(samples_or_fom, 'transfer_function'):
+            fom = samples_or_fom
+            if not isinstance(samples_or_fom, TransferFunction):
+                fom = fom.transfer_function
+            self.num_vars = 1 + fom.parameters.dim
+
+            assert len(sampling_values) == self.num_vars
+            self.parameters = fom.parameters
+            self.samples = np.empty([len(sv) for sv in sampling_values] + [fom.dim_output, fom.dim_input],
+                                    dtype=sampling_values[0].dtype)
+            for idx, vals in zip(np.ndindex(self.samples.shape[:-2]),
+                                 itertools.product(*sampling_values)):
+                params = fom.parameters.parse(vals[1:])
+                self.samples[idx] = fom.eval_tf(vals[0], mu=params)
+            if fom.dim_input == fom.dim_output == 1:
+                self.samples = self.samples.reshape(self.samples.shape[:-2])
+        else:
+            self.samples = samples_or_fom
+            self.num_vars = len(sampling_values)
+            self.parameters = {'p': self.num_vars-1}
+
+        # add complex conjugate samples
+        if conjugate:
+            s_conj_list = []
+            samples_conj_list = []
+            for i, s in enumerate(sampling_values[0]):
+                if s.conj() not in sampling_values[0]:
+                    s_conj_list.append(s.conj())
+                    samples_conj_list.append(self.samples[i, None].conj())
+            if s_conj_list:
+                sampling_values[0] = np.append(sampling_values[0], s_conj_list)
+                self.samples = np.vstack([self.samples] + samples_conj_list)
+
+        self.__auto_init(locals())
+
+    def reduce(self, tol=1e-7, itpl_part=None, max_itpl=None):
+        """Reduce using p-AAA.
+
+        Parameters
+        ----------
+        tol
+            Convergence tolerance for relative error of `rom` over the set of samples.
+        itpl_part
+            Initial partition for interpolation values. Should be `None` or a nested list
+            such that `itpl_part[i]` corresponds to indices of interpolated values with
+            respect to the `i`-th variable. I.e., `self.sampling_values[i][itpl_part[i]]`
+            represents a list of all initially interpolated samples of the `i`-th variable.
+            If `None` p-AAA will start with no interpolated values.
+        max_itpl
+            Maximum number of interpolation points to use with respect to each
+            variable. Should be `None` or a list such that `self.num_vars == len(max_itpl)`.
+            If `None` `max_itpl[i]` will be set to `len(self.sampling_values[i]) - 1`.
+
+        Returns
+        -------
+        rom
+            Reduced |TransferFunction| model.
+        """
+        svs = self.sampling_values
+        samples = self.samples
+
+        num_vars = len(svs)
```

Comment:
```suggestion
        num_vars = self.num_vars
```

Or `num_vars` usage below could be replaced by `self.num_vars`.